Entities/AtomGroup.cpp

```diff
@@ -1,2618 +1,1750 @@
-
-int AtomGroup::Create(MOSRotating *pOwnerMOSRotating)
-{
-    return Create(pOwnerMOSRotating, m_pMaterial, m_Resolution, m_Depth);
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          Create
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Creates a AtomGroup after the silhouette shape of a passed in MOSRotating,
-//                  by dotting the outline of the sprite with atoms. The passed in MOSRotating
-//                  will also be made the owner of this AtomGroup.
-
-int AtomGroup::Create(MOSRotating *pOwnerMOSRotating, Material const *material, int resolution, int depth)
-{
-    if (!pOwnerMOSRotating || resolution < 0)
-    {
-        RTEAbort("Trying to generate an AtomGroup without sprite and//or 0 resolution setting!");
-        return -1;
-    }
-
-    m_pOwnerMO = pOwnerMOSRotating;
-    m_AutoGenerate = true;
-    m_pMaterial = material;
-    m_Resolution = resolution > 0 ? resolution : g_MovableMan.GetAGResolution();
-    m_Resolution = m_Resolution > 0 ? m_Resolution : 1;
-    m_Depth = depth;
-    BITMAP *refSprite = pOwnerMOSRotating->GetSpriteFrame();
-	BITMAP *checkBitmap = 0;
-    int width = refSprite->w * static_cast<int>(pOwnerMOSRotating->GetScale());
-    int height = refSprite->h * static_cast<int>(pOwnerMOSRotating->GetScale());
-	int x;
-	int y;
-	int i;
-    Vector spriteOffset = pOwnerMOSRotating->GetSpriteOffset();
-    bool inside = false;
-
-
-// TODO: GET THIS WORKING WITH SCALED SPRITES!!$@#")
-    // This offset is to fix some weirdness going on with the rotozoom
-    int offX = -refSprite->w;
-    int offY = -refSprite->h;
-
-	// Only try to generate AtomGroup if scaled width and height are > 0 as we're playing with fire trying to create 0x0 bitmap. 
-	// In debug mode it fires an assertion fail somewhere in DirectX
-	if (width > 0 && height > 0)
-	{
-		checkBitmap = create_bitmap_ex(8, width, height);
-		clear_to_color(checkBitmap, g_MaskColor);
-
-		acquire_bitmap(refSprite);
-		acquire_bitmap(checkBitmap);
-
-		// If atoms are to be placed right at (below) the bitmap of the sprite.
-		if (depth <= 0)
-		{
-			// First scan horizontally, and place Atoms on outer silhouette edges.
-			for (y = 0; y < height; y += m_Resolution)
-			{
-				// Scan LEFT to RIGHT, placing one Atom on each first encountered silhouette edge
-				inside = false;
-				for (x = 0; x < width; ++x)
-				{
-					// Detect if we are crossing a silhouette boundary.
-					if (getpixel(refSprite, x, y) != g_MaskColor)
-					{
-						// Mark that an atom has been put in this location, to avoid duplicate Atoms
-						putpixel(checkBitmap, x, y, 99);
-						AddAtomToGroup(x, y, spriteOffset, pOwnerMOSRotating, true);
-						break;
-					}
-				}
+	int AtomGroup::ReadProperty(std::string propName, Reader &reader) {
+		if (propName == "Material") {
```

Comment:
Yeah ordering of ini properties isn't always going to match the header, they still need to make sense when being written out for the sake of not ass ini structure.
Writing all the Atoms before the other properties is just dumb and confusing in the ini.
It's just different sorting priorities I guess, in the header Atom is on top because like, AtomGroup consists of Atoms, so they're like the main thing so they're on top.
Also the order of `ReadProperty` depends on `Save`.

Reply:
Ohh yeah, fair enough about the writer spitting things out in a human-friendly format. I didn't think about that at all but that's a really good point.
That said, the order of ReadProperty doesn't actually depend on the order of Save at all, but I would agree that they should line up.
I'm okay with having these two not line up I guess, in cases where the human-friendly ordering for .ini files is different than for .h files.

Reply:
Yeah I know the read order doesn't *actually* depend on the write order to work, what I mean is that *I* order the read depending on the write.

Reply:
Yeah, I'm okay with that. I'd also be okay with just the write having different ordering cause it needs it and having the read match up with everything else, but as long as we're consistent I don't really mind either way.